src/safeposix/dispatcher.rs

```diff
@@ -1005,11 +1006,44 @@ pub fn lind_syscall_api(
                 .nanosleep_time64_syscall(clockid, flags, req, rem)
         }
 
+        SBRK_SYSCALL => {
+            let brk = arg1 as u32;
+
+            interface::sbrk_handler(cageid, brk)
+        }
+
         _ => -1, // Return -1 for unknown syscalls
     };
     ret
 }
 
+// initilize the vmmap, invoked by wasmtime
+pub fn lind_cage_vmmap_init(cageid: u64) {
+    let cage = interface::cagetable_getref(cageid);
+    let mut vmmap = cage.vmmap.write();
+    vmmap.add_entry(VmmapEntry::new(0, 0x30, PROT_WRITE | PROT_READ, 0 /* not sure about this field */, (MAP_PRIVATE | MAP_ANONYMOUS) as i32, false, 0, 0, cageid, MemoryBackingType::Anonymous));
+    // BUG: currently need to insert an entry at the end to indicate the end of memory space. This should be fixed soon so that
+    //      no dummy entries are required to be inserted
+    vmmap.add_entry(VmmapEntry::new(1 << 18, 1, PROT_NONE, 0 /* not sure about this field */, (MAP_PRIVATE | MAP_ANONYMOUS) as i32, false, 0, 0, cageid, MemoryBackingType::Anonymous));
+}
+
+// set the wasm linear memory base address to vmmap
+pub fn set_base_address(cageid: u64, base_address: i64) {
```

Comment:
can we set the base address in the init function?

Reply:
yes we could, but it requires a little bit more work on wasmtime part since `lind_cage_vmmap_init` and `set_base_address` are called at different place. Can we currently mark this as a TODO here? Since there are a lot of TODOs in wasmtime for memory stuff anyway, and I'll fix them all later

Reply:
sure